WebRTCPlugin/WebRTCProxy.cpp

```diff
@@ -8,321 +7,319 @@
+				}
+			}
+		}
+	};
+
+	//Create activeX object which is a
+	CComObject<RTCPeerConnection>* pc;
+	HRESULT hresult = CComObject<RTCPeerConnection>::CreateInstance(&pc);
+
+	if (FAILED(hresult))
+		return hresult;
+
+	//Create peerconnection object, it will call the AddRef inside as it gets a ref to the observer
+	rtc::scoped_refptr<webrtc::PeerConnectionInterface> pci = peer_connection_factory_->CreatePeerConnection(
+		configuration,
+		NULL,
+		NULL,
+		pc
+	);
+
+	//Check it was created correctly
+	if (!pci)
+		//Error
+		return E_INVALIDARG;
+
+	//Set event thread
+	pc->SetThread(eventThread);
+
+	//Attach to PC
+	pc->Attach(pci);
+
+	//Get Reference to pass it to JS
+	*peerConnection = pc->GetUnknown();
+
+	//Add JS reference
+	(*peerConnection)->AddRef();
+
+	//OK
+	return hresult;
 }
 
 
 STDMETHODIMP WebRTCProxy::createLocalAudioTrack(VARIANT constraints, IUnknown** track)
 {
-  const cricket::AudioOptions options;
-  //Create audio source
-  auto audioSource = peer_connection_factory_->CreateAudioSource(options);
+	const cricket::AudioOptions options;
+	//Create audio source
+	auto audioSource = peer_connection_factory_->CreateAudioSource(options);
+
+	//Ensure it is created
+	if (!audioSource)
+		return E_UNEXPECTED;
 
-  //Ensure it is created
-  if (!audioSource)
-    return E_UNEXPECTED;
+	//Create track
+	rtc::scoped_refptr<webrtc::MediaStreamTrackInterface> audioTrack = peer_connection_factory_->CreateAudioTrack("audio", audioSource);
 
-  //Create track
-  rtc::scoped_refptr<webrtc::MediaStreamTrackInterface> audioTrack = peer_connection_factory_->CreateAudioTrack("audio", audioSource);
+	//Ensure it is created
+	if (!audioTrack)
+		return E_UNEXPECTED;
 
-  //Ensure it is created
-  if (!audioTrack)
-    return E_UNEXPECTED;
+	//Create activeX object which is a
+	CComObject<MediaStreamTrack>* mediaStreamTrack;
+	HRESULT hresult = CComObject<MediaStreamTrack>::CreateInstance(&mediaStreamTrack);
 
-  //Create activeX object which is a
-  CComObject<MediaStreamTrack>* mediaStreamTrack;
-  HRESULT hresult = CComObject<MediaStreamTrack>::CreateInstance(&mediaStreamTrack);
+	if (FAILED(hresult))
+		return hresult;
 
-  if (FAILED(hresult))
-    return hresult;
-  
-  //Attach to native track
-  mediaStreamTrack->Attach(audioTrack);
+	//Attach to native track
+	mediaStreamTrack->Attach(audioTrack);
 
-  //Set dummy audio  label
-  mediaStreamTrack->SetLabel("Default Audio Device");
+	//Set dummy audio  label
+	mediaStreamTrack->SetLabel("Default Audio Device");
 
-  //Get Reference to pass it to JS
-  *track = mediaStreamTrack->GetUnknown();
+	//Get Reference to pass it to JS
+	*track = mediaStreamTrack->GetUnknown();
 
-  //Add JS reference
-  (*track)->AddRef();
+	//Add JS reference
+	(*track)->AddRef();
 
-  //OK
-  return hresult;
+	//OK
+	return hresult;
 }
 
+class CapturerTrackSource : public webrtc::VideoTrackSource
+{
+public:
+	static rtc::scoped_refptr<CapturerTrackSource> Create()
+	{
+		const size_t kWidth = 640;

```

Comment:
so the video is set hardcoded to 640x480 30fps? if so, you can set it as default but constraints should be able to be provided externally

Reply:
This example implementation of the new modulation to create a new video capture source is just that, an example that illustrates how it should work with the new changes in webRTC lib. You could change it however you see fit.